app/src/main/java/ch/epfl/sdp/artificial_intelligence/MovingEntity.java

```diff
@@ -0,0 +1,139 @@
+package ch.epfl.sdp.artificial_intelligence;
+
+import java.util.Random;
+
+public class MovingEntity implements Movable, Localizable, Updatable{
+    private GenPoint position;
+    private float velocity;
+    private float acceleration;
+    private float orientation;
+    private Movement movement;
+    private boolean moving;
+    private Boundable bounds;
+
+    public MovingEntity(){
+        position = new CartesianPoint(0,0);
+        acceleration = 0;
+        velocity = 0;
+        orientation = 0;
+        movement = Movement.LINEAR;
+        moving = false;
+        bounds = new UnboundedArea();
```

Comment:
Since we have the set and get, is it possible to change these variables to private

Reply:
Sorry, I don't understand your point. Those variables **are** private.

Reply:
Ah, I misread the line! You're right